### ISSUE TITLE
Feature request: projections

### ISSUE THREAD

**Issue by peter-conalgo** [josiahcarlson/rom#108]
I might be using the term "projection" wrong, but what I'd like is to be able to get a single column's values, based on a query, without loading the whole dataset. This API might be something like this:

```python
>>> Address.query.value('country_code')
```
Which would return either a list (or better, an iterator) for all countries in my set of addresses.

I'm happy with a single column projection. The extra API complexity to support multiple columns doesn't seem worth it to me.

**Comment by josiahcarlson** [comment]
In other ORMs, this is just a normal select call. No reason to complicate things here. You can use it like: `Address.query.select('country_code')`.

The actual prototype will look like:
`def select(self, *column_names, include_pk=True, decode=True, ff=_dict_data_factory):`

**Comment by josiahcarlson** [comment]
I'm building this right now, it will just be a method on the `Query` object or `Model.query` attribute.

**Comment by josiahcarlson** [comment]
I have this built and tested. Going to release it with a solution to issue #109 tomorrow.

**Comment by peter-conalgo** [comment]
Thanks for making this feature!

If I give it a single column (and `include_pk` False), will it give me bare values, or will they be wrapped in a dict or similar? I see the value of the complete feature set, but I think I'll end up unwrapping the values most times. Personally, I like how sqlalchemy does both `value()` and `values()`. 

Reference [sqlalchemy.orm.query.Query.value](http://docs.sqlalchemy.org/en/latest/orm/query.html?highlight=values#sqlalchemy.orm.query.Query.value)

**Comment by josiahcarlson** [comment]
You can pass your own factory function to return whatever you want. In this case, if you just want to pull the first column of data requested (regardless of how many there are):

```
def first_column_factory_function(columns):
    # ignore the columns
    def make(data):
        return data[0]
    return make
```

Then, if you want to get your unwrapped single column, you would:
```
for cv in Model.query.select('column', include_pk=False, ff=first_column_factory_function):
```

Here are a couple examples I have in my tests that show what you can do:
```
        # test no decoding
        total = 0
        for it in RomTestIterResult.query.select('col1', decode=False).order_by('col1').iter_result(30, 10):
            total += int(it['col1'])
        self.assertEqual(total, 50 * 51 / 2)

        from rom.query import _namedtuple_data_factory as ntf

        # test alternate factory function output with decoding
        total = 0
        for it in RomTestIterResult.query.select('col1', decode=True, ff=ntf).order_by('col1').iter_result(30, 10):
            total += it.col1
        self.assertEqual(total, 50 * 51 / 2)
```

**Comment by peter-conalgo** [comment]
Two things:

- I tried this feature today on a model that has 'required' fields, and if those fields aren't listed in the select, it doesn't work. (`rom.exceptions.MissingColumn: xxx.bbb cannot be missing`)

- Again, I would advocate for a `value()` and `values()` style interface. Those interfaces, even if they are just wrappers at this point, would permit future optimizations. For example, specialized Lua scripts could be written that are optimized for this path... someday.

**Comment by josiahcarlson** [comment]
Two things:
* That bug was only apparent when `decode=True` because decoding round-trips through the model. I've fixed the bug in the just-released 0.39.5 (we no longer raise an exception when model data is missing from Redis), added a test, and you shouldn't have any further issues.
* A specialized Lua path was already written that only fetches the specific data requested as part of the commit on June 27; remaining savings on the rom side of things amounts to 1 list slice per row.

**Comment by peter-conalgo** [comment]
Wonderful. As usual you are two steps ahead of me.
